### PR TITLE
Relocate Tests

### PR DESCRIPTION
**Pull request by plessbd**
The first commit is just a bunch of `git mv` folders.
```bash
git mv open_xdmod/modules/xdmod/regression_tests tests/regression
git mv open_xdmod/modules/xdmod/automated_tests/ tests/ui
git mv open_xdmod/modules/xdmod/component_tests/ tests/component
git mv open_xdmod/modules/xdmod/tests/ tests/unit
git mv open_xdmod/modules/xdmod/integration_tests/ tests/integration
git mv tests/xdmod-test-artifacts/xdmod tests/artifacts/xdmod
git mv tests/artifacts/xdmod/automated_tests tests/artifacts/xdmod/ui
```
The second is where the actual changing of files takes place.

The third commit is cleaning up the `artifacts/xdmod/` json files
```bash
for file in `find . -type f -name '*.json'`; do echo $file && jq . -- "$file" >| /tmp/cleaned.json && \mv -f /tmp/cleaned.json "$file"; done
```

fourth commit fixes one last travis error.

`tests` is a wonderful directory for things such as regression, integration, ui, and unit tests to be located.

so I moved them there

`automated_tests` is a dumb name (thanks @plessbd) these have been renamed to `ui`
no need to have `_tests` in the folder names anymore since these are now properly located in a test directory

`tests/ci`
-  `.secrets.json` isnt a secret to anyone so created that under here so it doesnt ahve to be copied around
-  the scripts needed for ci are now here instead of under integration_tests/scripts

### CODE REVIEW

File: tests/component/lib/Utilities/LogAndThrowExceptionTest.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ public function testNoExceptionCode()
     public function testExceptionCode()
     {
         $msg = "Code = 10";
-        $excpectedMsg = (string) $this->loggable . " $msg";
+        $expectedMsg = (string) $this->loggable . " $msg";
```
Comment (jpwhite4) [inline tests/component/lib/Utilities/LogAndThrowExceptionTest.php]: How did this test pass before?

File: tests/integration/lib/Controllers/BaseUserAdminTest.php [hunk 2]
```diff
@@ -210,7 +210,7 @@ protected function createUser(array $options)
         $institution = isset($options['institution']) ? $options['institution'] : -1;
         $firstName = isset($options['first_name']) ? $options['first_name'] : 'Test';
         $lastName = isset($options['last_name']) ? $options['last_name'] : 'User';
-        $emailAddress = isset($option['email_address']) ? $options['email_address'] : $username . self::DEFAULT_EMAIL_ADDRESS_SUFFIX;
+        $emailAddress = isset($options['email_address']) ? $options['email_address'] : $username . self::DEFAULT_EMAIL_ADDRESS_SUFFIX;
```
Comment (jpwhite4) [inline tests/integration/lib/Controllers/BaseUserAdminTest.php]: How did this test pass before?

File: tests/ui/wdio.conf.js [hunk 3]
```diff
@@ -218,7 +218,7 @@ exports.config = {
             if (webdriverHelpers.hasOwnProperty(extension) && typeof webdriverHelpers[extension] === 'function') {
                 browser.addCommand(extension, webdriverHelpers[extension].bind(browser));
             } else {
-                console.warning('Not Adding: ', extension, typeof webdriverHelpers[extension]);
+                process.stderr.write('Not Adding: ' + extension + typeof webdriverHelpers[extension]);
```
Comment (jpwhite4) [inline tests/ui/wdio.conf.js]: Note that this code will produce different output to the previous code.

File: tests/ci/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+docker build -t centos7:xdmod-base -f Dockerfile.centos7:xdmod-base .
+```
+
+## Cache and Config Files
+We use some default configuration and cache files to set some defaults for
+development and speed up the build times of Docker.
+If you look in `assets/copy-caches.sh` you will see all of the files that will be put into place.
+
+### vimrc
+This is just some defaults for VIM when useing the docker for development
```
Comment (jpwhite4) [inline tests/ci/README.md]: s/useing/using/

File: tests/component/lib/Utilities/LogAndThrowExceptionTest.php (hunk 1 above)
Comment (plessbd) [reply tests/component/lib/Utilities/LogAndThrowExceptionTest.php]: Magic?

File: tests/integration/lib/Controllers/BaseUserAdminTest.php (hunk 2 above)
Comment (plessbd) [reply tests/integration/lib/Controllers/BaseUserAdminTest.php]: Magic?

File: tests/ui/wdio.conf.js (hunk 3 above)
Comment (plessbd) [reply tests/ui/wdio.conf.js]: yes, however, this is for the most part only for "developers" when running this and should rearely if ever be hit.

**Pull request merged**